app/controllers/user/swaps_controller.rb

```diff
@@ -66,14 +74,14 @@ def assert_mobile_phone_verified
   def assert_has_email
     return unless @user.email.blank?
 
-    flash[:errors] = ["Please enter your email address before you swap!"]
+    flash[:errors] = ["Please enter your email address before you swap"]
```

Comment:
These should end in `.` too, no? But not a merge-blocker.

Reply:
No it probably shouldn't. I would generally end with a "." if it's multiple sentences but not otherwise.